### PR TITLE
Bugfix/table sortable datesortcolumnindex fix

### LINKED ISSUE #1765: Feature 'datesortcolumnindex' of axa-table-sortable doesn't work
Hi everyone
Dates are sorted like a number, even with the new attribute 'datesortcolumnindex'. Noah Liechti thinks, that it is a bug.

Here is my code:

the definition of the sortable table feature 
```javascript
<AXATableSortable model={model} datesortcolumnindex="6" maxheight="0" innerscroll="0" onClick={(e) => {
    stores[0].tableMenuShow(e)
}}/>
```

The content factory of the table:

```javascript
for (let y = 0; y < anzahlSpalten; y++) {
    let datumLimite = '';
    let datumFormat = {year: 'numeric', month: '2-digit', day: '2-digit'};
    if (stores[0].state.limiten[y].climGab.trim() !== '') {
        datumLimite = new Intl.DateTimeFormat('de-CH', datumFormat).format(new Date(stores[0].state.limiten[y].climGab));
    }
    let status = 'offen';
    if (stores[0].state.statusTitles.length !== 0) {
        if (stores[0].state.limiten[y].climLTC > 0 && stores[0].state.limiten[y].climLTC < 100) {
            status = stores[0].state.statusTitles[guiSprache][0];
        }
        if (stores[0].state.limiten[y].climLTC > 99 && stores[0].state.limiten[y].climLTC < 200) {
            status = stores[0].state.statusTitles[guiSprache][1];
        }
        if (stores[0].state.limiten[y].climLTC > 199 && stores[0].state.limiten[y].climLTC < 300) {
            status = stores[0].state.statusTitles[guiSprache][2];
        }
        if (stores[0].state.limiten[y].climLTC > 299 && stores[0].state.limiten[y].climLTC < 400) {
            status = stores[0].state.statusTitles[guiSprache][3];
        }
        if (stores[0].state.limiten[y].climLTC > 399 && stores[0].state.limiten[y].climLTC < 500) {
            status = stores[0].state.statusTitles[guiSprache][4];
        }
        if (stores[0].state.limiten[y].climLTC > 499 && stores[0].state.limiten[y].climLTC < 600) {
            status = stores[0].state.statusTitles[guiSprache][5];
        }
        if (stores[0].state.limiten[y].climLTC > 599 && stores[0].state.limiten[y].climLTC < 700) {
            status = stores[0].state.statusTitles[guiSprache][6];
        }
        if (stores[0].state.limiten[y].climLTC > 699 && stores[0].state.limiten[y].climLTC < 800) {
            status = stores[0].state.statusTitles[guiSprache][7];
        }
        if (stores[0].state.limiten[y].climLTC > 799 && stores[0].state.limiten[y].climLTC < 900) {
            status = stores[0].state.statusTitles[guiSprache][8];
        }
    }
    let land = "";
    if (stores[0].state.limiten[y].climAdresse[0].claendercode.cLaendercodeAlt !== '') {
        land = stores[1].state.laenderAuswahl.find(item => item.cLaendercodeAlt === stores[0].state.limiten[y].climAdresse[0].claendercode.cLaendercodeAlt)
    } else {
        land = stores[1].state.laenderAuswahl.find(item => item.cLaendercode === stores[0].state.limiten[y].climAdresse[0].claendercode.cLaendercode)
    }
    if (land === "" || land === undefined) {
        land = stores[0].state.limiten[y].climAdresse[0].claendercode;
    }
    let zeile = [
        {html: '<span style="float: right">' + stores[0].state.limiten[y].climDossierNr + ''},
        {html: '<span>' + stores[0].state.limiten[y].climAdresse[0].adresseAllinOne + '</span>'},
        {html: '<span>' + land.cLandname[landInSprache] + '</span>'},
        {html: '<span>' + stores[0].state.limiten[y].climRef + '</span>'},
        {html: '<span style="float: right">' + new Intl.NumberFormat('de-CH').format(stores[0].state.limiten[y].climGLimEW * 1000) + '</span>'},
        {html: '<span style="float: right">' + new Intl.NumberFormat('de-CH').format(stores[0].state.limiten[y].climLimite * 1000) + '</span>'},
        {html: '<span style="float: right">' + datumLimite + '</span>'},
        {html: '<span>' + status + '</span>'},
    ];
    anzahlAktuelleSpalten++;
    tbody.push(zeile);
}
```

The building of the table model:
```javascript
let model = {
    thead: [
        {html: headlineDrNumber, sort: 'DESC'},
        {html: headlineBuyer, sort: 'ASC'},
        {html: headlineCountry, sort: 'ASC'},
        {html: headlineReference, sort: 'ASC'},
        {html: headlineLimitDesired, sort: 'ASC'},
        {html: headlineLimitApproved, sort: 'ASC'},
        {html: headlineValidFrom, sort: 'ASC'},
        {html: headlineStatus, sort: 'ASC'},
    ],
    tbody: tbody,
};
```

### PR DESCRIPTION
**Pull request by 01231**
Fixes #1765

# Done is when (DoD):
- [ ] Modifications within components are reflected by tests.
- [x] A self-review of the code has been performed.
- [x] The modified component properties have been added to the typescript typings.
- [x] Potential design changes have been reviewed by a designer.
- [x] The modifications are well documented (README.md, etc.) and showcased in the stories.
- [x] The modifications are documented in the code, particularly in hard-to-understand areas (focus on **WHY**).
- [x] The modifications are shortly added to CHANGELOG.md with the issue number.
- [x] The modifications still work in IE.
- [x] The modifications are according to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)

### CODE REVIEW

File: src/components/30-organisms/table-sortable/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.1.3
+
+- `datesortcolumnindex` now works with react. #1765
```
Comment (markus-walther) [inline src/components/30-organisms/table-sortable/CHANGELOG.md]: Capitalize: React

**Review by markus-walther — CHANGES_REQUESTED** [review]
LGTM. Question: what's the specific part that was failing under React? I suggest to identify/explain with a comment so we can learn something reading your source code ;-)

File: src/components/30-organisms/table-sortable/CHANGELOG.md (hunk 1 above)
Comment (01231) [reply src/components/30-organisms/table-sortable/CHANGELOG.md]: Thank you!

**Comment by 01231** [comment]
@markus-walther the problem was that the value of `dateSortColumnIndex` (`this.dateSortColumnIndex`) wasn't available in the constructor. The reason for that I didn't find out.

**Review by danyball — APPROVED** [review]
lgtm

**Pull request merged**